cmd/agent/subcommands/snmp/command.go

```diff
@@ -82,219 +75,160 @@ type cliParams struct {
-	snmpWalkCmd.Flags().IntVarP(&cliParams.timeout, "timeout", "t", defaultTimeout, "Set the request timeout (in seconds)")
-	snmpWalkCmd.Flags().BoolVar(&cliParams.unconnectedUDPSocket, "use-unconnected-udp-socket", defaultUseUnconnectedUDPSocket, "If specified, changes net connection to be unconnected UDP socket")
-
-	snmpWalkCmd.SetArgs([]string{})
+// usageError wraps any error that indicates the user has provided invalid inputs.
+// If the main script returns a usageError it will print the usage string along
+// with the error message.
+type usageError struct {
+	e error
+}
 
-	snmpCmd := &cobra.Command{
-		Use:   "snmp",
-		Short: "Snmp tools",
-		Long:  ``,
+func (u usageError) Error() string {
+	if u.e != nil {
+		return u.e.Error()
 	}
-	snmpCmd.AddCommand(snmpWalkCmd)
+	return "Usage error"
+}
 
-	return []*cobra.Command{snmpCmd}
+func (u usageError) Unwrap() error {
+	return u.e
 }
 
-//nolint:revive // TODO(NDM) Fix revive linter
-func snmpwalk(config config.Component, cliParams *cliParams) error {
-	var (
-		address      string
-		deviceIP     string
-		oid          string
-		port         uint16
-		value        uint64
-		setVersion   gosnmp.SnmpVersion
-		authProtocol gosnmp.SnmpV3AuthProtocol
-		privProtocol gosnmp.SnmpV3PrivProtocol
-		msgFlags     gosnmp.SnmpV3MsgFlags
-	)
-	// Get args
-	if len(cliParams.args) == 0 {
-		fmt.Print("Missing argument: IP address\n")
-		cliParams.cmd.Help() //nolint:errcheck
-		os.Exit(1)
-		return nil
-	} else if len(cliParams.args) == 1 {
-		address = cliParams.args[0]
-		oid = defaultOID
-	} else if len(cliParams.args) == 2 {
-		address = cliParams.args[0]
-		oid = cliParams.args[1]
-	} else {
-		fmt.Printf("The number of arguments must be between 1 and 2. %d arguments were given.\n", len(cliParams.args))
-		cliParams.cmd.Help() //nolint:errcheck
-		os.Exit(1)
-		return nil
-	}
-	if strings.Contains(address, ":") {
-		deviceIP = address[:strings.Index(address, ":")]
-		value, _ = strconv.ParseUint(address[strings.Index(address, ":")+1:], 0, 16)
-		port = uint16(value)
-	} else {
-		//If the customer provides only 1 argument : the ip_address
-		//We check the ip address configuration in the agent runtime and we use it for the snmpwalk
-		deviceIP = address
-		//Allow the possibility to pass the config file as an argument to the command
+// usagef is a shorthand for creating a simple usageError.
+func usagef(msg string, args ...any) usageError {
+	return usageError{fmt.Errorf(msg, args...)}
+}
+
+// getParams tries to parse deviceAddr as host:port; if it can, it just returns
+// (host, port, params, nil). If the address is not a host:port pair, then it
+// assumes it is a plain IP address, and queries the running agent for the
+// configuration for that address.
+func getParams(deviceAddr string, connParams *connectionParams) (string, uint16, *connectionParams, error) {
+	deviceIP, port, hasPort := parseIPWithMaybePort(deviceAddr)
+
+	if !hasPort {
```

Comment:
this check should be on the number of the arguments and not the port, if we don't have a port ut have other arguments then we should use the default port and the given arguments for the command and not check the config from the agent

Reply:
I have updated it to always check the config and use the config values for anything that is unset, but use the user-provided values for anything the user provided.

If it can't read config from the agent, or the agent has no config for the given IP, it prints a warning message but continues anyway.